fn_phish_tank/fn_phish_tank/components/fn_phish_tank_submit_url.py

```diff
@@ -0,0 +1,92 @@
+# (c) Copyright IBM Corp. 2010, 2019. All Rights Reserved.
+# -*- coding: utf-8 -*-
+# pragma pylint: disable=unused-argument, no-self-use
+"""Function implementation"""
+
+import logging
+from resilient_circuits import ResilientComponent, function, handler, StatusMessage, FunctionResult, FunctionError
+import fn_phish_tank.util.selftest as selftest
+from resilient_lib import ResultPayload
+from fn_phish_tank.util.phish_tank_helper import *
+
+
+class FunctionComponent(ResilientComponent):
+    """Component that implements Resilient function 'fn_phish_tank_submit_url"""
+
+    def __init__(self, opts):
+        """constructor provides access to the configuration options"""
+        super(FunctionComponent, self).__init__(opts)
+        self.options = opts.get("fn_phish_tank", {})
+        selftest.selftest_function(opts)
+
+    @handler("reload")
+    def _reload(self, event, opts):
+        """Configuration options have changed, save new values"""
+        self.options = opts.get("fn_phish_tank", {})
+
+    @function("fn_phish_tank_submit_url")
+    def _fn_phish_tank_submit_url_function(self, event, *args, **kwargs):
+        """Function: Function to check URLs against PhishTanks(https://www.phishtank.com/) Database to see URL is flagged as phishing or not phishing."""
+        try:
+            # Get the function parameters:
+            phish_tank_check_url = kwargs.get("phish_tank_check_url")  # text
+
+            # Get function parameters from config file.
+            phish_tank_api_url = self.options.get('phish_tank_api_url')
+            phish_tank_api_key = self.options.get('phish_tank_api_key')
+            proxy = self.options.get('proxy')
+
+            log = logging.getLogger(__name__)
+            log.info("PhishTank Check URL: %s", phish_tank_check_url)
+            log.info("PhishTank API Access URL: %s", phish_tank_api_url)
+            log.info("PhishTank API Access KEY: %s", phish_tank_api_key)
+            log.info("Proxy Server Address: %s", proxy)
+
+            yield StatusMessage("Checking Phishing Status Against URL: {}".format(phish_tank_check_url))
+
+            # Initialing the resilient result object
+            _result_obj = ResultPayload('fn_phish_tank',
+                                        function_inputs=[phish_tank_api_url, phish_tank_check_url, phish_tank_api_key])
+
+            # PhihshTank Proxy Data
+            pt_proxy = phish_tank_helper.format_proxy_data(proxy_data=proxy)
+
+            # PhishTank POST Call Data
+            pt_post_call_data = phish_tank_helper.create_post_data(phish_tank_check_url, api_key=phish_tank_api_key)
+
+            # Initialising the Request Session
+            pt_helper_class_instance = phish_tank_helper()
+            _request_session = pt_helper_class_instance.session()
```

Comment:
Are you closing `_request_session` ? 
I can see you are closing `_api_response` but not `_request_session`

Reply:
connection is available on the response object, so closing on `_api_response`

Reply:
I think you are right on this, great job!